Set GAP_DIR and GAP_PRG in the startup script gap.sh.

* GAP_DIR is set using the SAGE_LOCAL environment variable

* GAP_PRG is determined at runtime based on `uname` and `uname -m`

diff -ur src/bin/gap.sh.orig src/bin/gap.sh
--- src/bin/gap.sh.orig	2015-07-24 07:51:10.000000000 -0400
+++ src/bin/gap.sh	2015-07-24 07:51:42.000000000 -0400
@@ -8,18 +8,21 @@
 ##  Copy/rename this file to a directory in your search path like '~/bin/gap'.
 ##  If you later move GAP to another location you must only change this file.
 ##
+if [ $SAGE_LOCAL"" = "" ]; then
+    echo "Set the environment variable SAGE_LOCAL."
+    exit 1
+fi
 
 #############################################################################
 ##
 ##  GAP_DIR . . . . . . . . . . . . . . . . . . . . directory where GAP lives
 ##
-##  Set 'GAP_DIR' to the name of the directory where you have installed  GAP,
-##  i.e., the directory with the subdirectories  'lib',  'grp',  'doc',  etc.
-##  The default is '/usr/local/lib/gap3-jm5'.
-##  You have to change this unless you have installed  GAP in this  location.
+##  This script tries to determine the  directory  where  you  installed  GAP
+##  automatically, but if it doesn't work, then set 'GAP_DIR' to the name  of
+##  the directory where you have installed GAP, i.e., the directory with  the
+##  subdirectories 'lib', 'grp', 'doc', etc.
 ##
-#GAP_DIR=/usr/local/lib/gap3-jm5
-GAP_DIR=/home/jmichel/gap3-jm5
+GAP_DIR="$SAGE_LOCAL/gap3/latest"
 
 #############################################################################
 ##
@@ -42,9 +45,24 @@
 ##        gap.x86linux for linux X86
 ##        gap.x86linux64 for linux X86, 64 bit gap
 ##        gap.mac for Mac OSX on powerPC
-##        gap.macx86 for Mac OSX on X86 
+##        gap.macx86 for Mac OSX on X86
+##  The default setting is to determine this automatically at runtime.
 ##
-GAP_PRG=gap.x86linux64
+
+UNAME=$(uname)
+if [[ $UNAME == 'Linux' ]]; then
+    if [[ $(uname -m) == 'x86_64' ]]; then
+        GAP_PRG=gap.x86linux64
+    else
+        GAP_PRG=gap.x86linux
+    fi;
+elif [[ $UNAME == 'Darwin' ]]; then
+    if [[ $(uname -m) == 'x86_64' ]]; then
+        GAP_PRG=gap.macx86
+    else
+        GAP_PRG=gap.mac
+    fi;
+fi
 
 #############################################################################
 ##
